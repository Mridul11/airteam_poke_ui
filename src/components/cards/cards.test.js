import { render, screen } from '@testing-library/react';
import Cards from '.';
import MAINIMAGE from '../../assets/images/main.jpeg';
import AVATARIMAGE from '../../assets/images/avatar.jpeg';
import { MemoryRouter } from 'react-router-dom';
import AddWatchMedia from '../../utils/test-utils/add-matchmedia';

describe('cards component', () => {
  beforeAll(AddWatchMedia);
  test('renders cards element', () => {
    render(
      <MemoryRouter>
        <Cards
          name={'Name'}
          url={'https://pokeapi.co/api/v2/pokemon'}
          index={0}
          mainImage={MAINIMAGE}
          avatarImage={AVATARIMAGE}
        />
      </MemoryRouter>
    );
    const cardElement = screen.getByTestId('test-cards');
    expect(cardElement).toMatchSnapshot();
  });
});
