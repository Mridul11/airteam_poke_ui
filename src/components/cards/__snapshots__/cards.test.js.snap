// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cards component renders cards element 1`] = `
<div
  class="cards-spacing"
  data-testid="test-cards"
>
  <div
    class="ant-card ant-card-bordered"
    data-testid="test-card"
  >
    <div
      class="ant-card-cover"
    >
      <img
        alt="example"
        src="main.jpeg"
      />
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-meta"
      >
        <div
          class="ant-card-meta-avatar"
        >
          <span
            class="ant-avatar ant-avatar-circle ant-avatar-image"
          >
            <img
              src="avatar.jpeg"
            />
          </span>
        </div>
        <div
          class="ant-card-meta-detail"
        >
          <div
            class="ant-card-meta-title"
          >
            NAME
          </div>
          <div
            class="ant-card-meta-description"
          >
            <button
              class="ant-btn ant-btn-default"
              data-testid="test-knowmorebtn"
              type="button"
            >
              <a
                href="/about/0"
              >
                Know More
              </a>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
